aeron-driver/src/main/c/aeron_ipc_publication.c

```diff
@@ -449,6 +455,32 @@ void aeron_ipc_publication_on_time_event(
             break;
         }
 
+        case AERON_IPC_PUBLICATION_STATE_REVOKED:
+        {
+            aeron_driver_conductor_publication_transition_to_revoked(conductor, &publication->conductor_fields.managed_resource);
+
+            for (size_t i = 0, size = conductor->ipc_subscriptions.length; i < size; i++)
+            {
+                aeron_subscription_link_t *link = &conductor->ipc_subscriptions.array[i];
+
+                if (aeron_driver_conductor_is_subscribable_linked(
+                    link,
+                    &publication->conductor_fields.subscribable))
+                {
+                    aeron_driver_conductor_on_unavailable_image(
+                        conductor,
+                        publication->conductor_fields.managed_resource.registration_id,
+                        link->registration_id,
+                        publication->stream_id,
+                        AERON_IPC_CHANNEL,
+                        AERON_IPC_CHANNEL_LEN);
+                }
+            }
+
+            publication->conductor_fields.state = AERON_IPC_PUBLICATION_STATE_DRAINING;
+            break;
+        }
+
         case AERON_IPC_PUBLICATION_STATE_DRAINING:
```

Comment:
In the revoked case, I'm not sure I'm skipping through here quick enough - i.e. for IPC we're still going through the DRAINING state.